packages/extension-manager/src/common/extension-protocol.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+
+import { JsonRpcServer } from "@theia/core";
+
+export interface ExtensionIndetifier {
+    readonly name: string
+    readonly version: string;
+}
+
+export interface Extension extends ExtensionIndetifier {
+    readonly description: string;
+    readonly repository: string;
+    readonly author: string;
+    readonly installed: boolean;
+    readonly outdated: boolean;
```

Comment:
outdated is after https://docs.npmjs.com/cli/outdated, actually we probably just should use this command